src/app/main/time-groups/edit-time-group/edit-time-group.component.ts

```diff
@@ -0,0 +1,240 @@
+        this.timeGroup.time_group_schedule_attributes = response.time_group_schedule;
+        delete this.timeGroup.time_group_schedule_attributes?.schedule_days;
+        delete this.timeGroup.time_group_schedule;
+      }
+      this.timeGroupLoading = false;
+
+    })
+  }
+
+  public openTimeScheduleModal(employee: Employee) {
+    let dialogRef = this.dialog.open(TimeScheduleComponent, {
+      width: '750px',
+      disableClose: true,
+      data: {
+        employeeId: employee.id!,
+        timeGroupId: this.timeGroupId,
+        employeeTimeSchedule: employee.time_group_schedule!,
+        timeScheduleLoading: false,
+        timeGroupEmployeesIds: this.timeGroup.employees?.map(employee => { return employee.id })
+      }
+    });
+    dialogRef.afterClosed().subscribe(result => {
+      if (result == 'update') {
+        this.getTimeGroup(this.timeGroupId);
+      }
+    })
+  }
+
+  public updatetimeGroupScheduleDays(scheduleDays: TimeGroupSchedule[]) {
+    this.timeGroup.time_group_schedule_attributes!.schedule_days_attributes != scheduleDays;
+  }
+
+  public updateTimeGroup() {
+    this.isFormSubmitted = true;
+    let isValidDays: boolean[] = []
+    if (this.timeGroup.group_type == 'fixed') {
+      this.timeGroup.time_group_schedule_attributes!.schedule_days_attributes!.forEach(day => {
+        if (day.is_off) {
+          day.clock_in = null;
+          day.clock_out = null;
+        }
+
+        if ((day.clock_in == '' || day.clock_out == '') && !day.is_off) {
+          day.invalidTime = true;
+          isValidDays.push(true)
+        }
+      })
+
+    }
+    else {
+      delete this.timeGroup.time_group_schedule_attributes
+    }
+
+    if (this.timeGroup.name == '' || isValidDays.includes(true)) {
+      this.isFormSubmitted = false;
+      this.invalidAllDaysTime = isValidDays.includes(true)
+      return
+    }
+    this.timeGroup.rule_id = this.timeGroup.rule?.id
+    this.timeGroupService.editTimeGroup(this.timeGroupId, { time_group: this.timeGroup }).subscribe(response => {
+      this.appNotificationService.push(this.translate.instant('tr_time_group_updated_successfully'), 'success');
+      this.isFormSubmitted = false;
+    }, error => {
+      this.appNotificationService.push(error.error.name, 'error');
+      this.isFormSubmitted = false;
+    })
+  }
+
+  public getDepartments() {
+    this.departmentsLoading = true
+    this.subscriptions.push(this.employeesService.getDepartments(this.dePartmentsParams).subscribe((response: any) => {
+      this.departmentsPagination = response.meta;
+      this.departments = this.departments.concat(response.employee_departments);
+      this.departmentsLoading = false
+    }))
+  }
+  public getRules() {
+    this.rulesLoading = true;
+    this.subscriptions.push(this.rulesSerivce.getRules(this.rulesParams).subscribe((response: any) => {
+      this.rules = response.rules;
+      this.rulesPagination = response.meta;
+      this.rulesLoading = false;
+
+    }))
```

Comment:
Please handle errors for the remaining of calls

Reply:
done